torchgeo/transforms/indices.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""TorchGeo indices transforms."""
+
+from typing import Dict
+
+import torch
+from torch import Tensor
+from torch.nn import Module  # type: ignore[attr-defined]
+
+# https://github.com/pytorch/pytorch/issues/60979
+# https://github.com/pytorch/pytorch/pull/61045
+Module.__module__ = "torch.nn"
+
+
+def ndvi(red: Tensor, nir: Tensor) -> Tensor:
+    """Compute Normalized Different Vegetation Index (NDVI).
+
+    Args:
+        red: tensor containing red band
+        nir: tensor containing nir band
+
+    Returns:
+        tensor containing computed NDVI values
+    """
+    return (nir - red) / (nir + red)
+
+
+class NDVI(Module):  # type: ignore[misc,name-defined]
+    """Normalized Difference Vegetation Index (NDVI).
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1807.05713
+    """
+
+    def __init__(self, index_red: int, index_nir: int) -> None:
+        """Initialize a new transform instance.
+
+        Args:
+            index_red: index of the Red band in the image
+            index_nir: index of the Near Infrared (NIR) band in the image
+        """
+        super().__init__()
+        self.index_red = index_red
+        self.index_nir = index_nir
+
+    def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
```

Comment:
Do the other Kornia transforms take in a dict or a Tensor?

Reply:
`AugmentationSequential` unfortunately takes in `n` tensors as the args in the `forward` method and uses the `data_keys` to identify them. We will likely need a wrapper to make it work with out dict. IMO it would make more sense to accept `data_keys` as a parameters and then take as input a dict and only modify the `data_keys`. Perhaps I should make an issue on the kornia repo about this.

Example:

```python
import torch
import kornia.augmentation as K

transforms = K.AugmentationSequential(
  K.RandomResizedCrop(size=(256, 256)),
  K.ColorJitter(),
  data_keys=["input", "mask"]
)
batch = dict(
 image=torch.randn(2, 3, 256, 256),
 mask=torch.zeros(2, 256, 256)
)
batch["image"], batch["mask"] = transforms(batch["image"], batch["mask"])
```